gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/translator/DotNetTranslator.java

```diff
@@ -0,0 +1,434 @@
+
+                    // have to special case withSack() because UnaryOperator and BinaryOperator signatures
+                    // make it impossible for the interpreter to figure out which function to call. specifically we need
+                    // to discern between:
+                    //     withSack(A initialValue, UnaryOperator<A> splitOperator)
+                    //     withSack(A initialValue, BinaryOperator<A> splitOperator)
+                    // and:
+                    //     withSack(Supplier<A> initialValue, UnaryOperator<A> mergeOperator)
+                    //     withSack(Supplier<A> initialValue, BinaryOperator<A> mergeOperator)
+                    if (methodName.equals(TraversalSource.Symbols.withSack) &&
+                            instruction.getArguments().length == 2 && instruction.getArguments()[1] instanceof Lambda) {
+                        final String castFirstArgTo = instruction.getArguments()[0] instanceof Lambda ? "ISupplier" : "";
+                        final Lambda secondArg = (Lambda) instruction.getArguments()[1];
+                        final String castSecondArgTo = secondArg.getLambdaArguments() == 1 ? "IUnaryOperator" : "IBinaryOperator";
+                        if (!castFirstArgTo.isEmpty())
+                            script.append(String.format("(%s) ", castFirstArgTo));
+                        convertToScript(instruction.getArguments()[0]);
+                        script.append(", (").append(castSecondArgTo).append(") ");
+                        convertToScript(instruction.getArguments()[1]);
+                        script.append(",");
+                    } else {
+                        for (final Object object : instruction.getArguments()) {
+                            // overloads might have trouble with null. add more as we find them i guess
+                            if (null == object && methodName.equals(GraphTraversal.Symbols.addV))
+                                script.append("(string) ");
+                            convertToScript(object);
+                            script.append(",");
+                        }
+                    }
+                    script.setCharAtEnd(')');
+                }
+            }
+            return script;
+        }
+
+        @Override
+        protected Script produceScript(final P<?> p) {
+            if (p instanceof TextP) {
+                script.append("TextP.").append(SymbolHelper.toCSharp(p.getBiPredicate().toString())).append("(");
+                convertToScript(p.getValue());
+            } else if (p instanceof ConnectiveP) {
+                final List<P<?>> list = ((ConnectiveP) p).getPredicates();
+                for (int i = 0; i < list.size(); i++) {
+                    produceScript(list.get(i));
+                    if (i < list.size() - 1) {
+                        script.append(p instanceof OrP ? ".Or(" : ".And(");
```

Comment:
The bracket from `Or(` / `And(` isn't closed anywhere if I'm not mistaken. The output at least has a a missing bracket when I combine multiple _Ands_:

```groovy
gremlin> translator.translate(g.V().has('age', gt(1).and(gt(2)).and(gt(3)))).getScript()
==>g.V().Has("age",P.Gt(1).And(P.Gt(2).And(P.Gt(3)))
```

Reply:
another hole in the test suite

Reply:
fixed this issue and added a test to ensure it's tested going forward.